### PR TITLE
test(base): add preview observer tests

### PR DESCRIPTION
**Pull request by ricokahler**
### Description

This PR adds some test coverage for the preview observer. This was done on top of #2585 to ensure all the necessary cases were still working.

Note: these tests are integration tests because they actually call the API.

CleanShot 2021-07-02 at 20 23 40@2x

### What to review

Ensure this is the right approach and everything looks good 😁

I also updated some of `observeFields` in order to create some small hooks for closing and cleaning up after tests.

Update: Looks like the Cypress token used here does not have permission to create docs. We probably need to add another for this suite.

### CODE REVIEW

**Comment by ricokahler** [comment]
Change [this line](https://github.com/sanity-io/sanity/pull/2602/files#diff-f70f02befcfe9d427bc931f640c62eac6f3a82173c6fcfeb5ce1152e2e3a63c4R5) and add in a token with create permissions to make these tests pass.

**Review by bjoerge — APPROVED** [review]
Really great work, Rico! 👏🏼 

General feeback (none of which are blockers for merge, but rather points for consideration :)

- Would it be possible to mock the sanity client entirely so that we don't have to rely on making http calls from the test runner? It would also make the tests less reliant on external factors and arbitrary server timings, and make it easier to assert the behavior of the preview observers in isolation. (this may be hard to achieve as it is now, but a goal nevertheless). As it is now these tests are somewhere in between integration tests and a unit tests, which is a great start. Edit: I see now that you mentioned these were integration tests, but considering what they cover I still think it's a good future improvement to make them into unit tests by not having to rely on external api calls.

- I think we have grown to prefer the convention of having unit tests alongside the source code (e.g. in a `__test__` subfolder). I know that's not the case in the base package yet, but if not too much work it would be nice to start adopting this convention here as well.

**Comment by bjoerge** [comment]
Regarding token, I think you can safely add a separate editor token for these tests (instead of re-using the cypress token) and add it as a workflow secret here: https://github.com/sanity-io/sanity/settings/secrets/actions

File: packages/@sanity/base/test/observeFields.test.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      _id: id,
+      _type: 'tempTestDocObserveFields',
+      name: 'testing',
+      otherValue: 'foo',
+    }
+
+    await client.createOrReplace(testDoc)
+
+    // type Unwrap<T> = T extends Observable<infer U> ? U : T
+    // const collectedValues: Unwrap<ReturnType<typeof observeFields>>[] = []
```
[the remark this answers is not in the document]
Comment (ricokahler) [reply packages/@sanity/base/test/observeFields.test.ts:64]: Unintentional 😇

**Comment by rexxars** [comment]
What is the status of this PR?

**Comment by ricokahler** [comment]
i think i'll close this but re-address soon

**Pull request closed without merging**